test/unit/orchestration/templates_test.rb

```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+
+class TemplatesTest < ActiveSupport::TestCase
+  setup do
+    @host = FactoryBot.build(:host, :managed)
+    @host.stubs(:skip_orchestration_for_testing?).returns(false)
+    @queue = @host.queue
+  end
+
+  test 'without template' do
+    @host.queue_render_checks
+    assert_equal 0, @queue.count
+  end
+
+  test 'with valid provisioning template' do
+    template = FactoryBot.build(:provisioning_template, template: "<%= @host.name %>")
+    @host.stubs(:provisioning_template).returns(template)
+    @host.queue_render_checks
+    assert_equal 1, @queue.count
+    assert @host.set_renderability
+  end
+
+  test 'with invalid provisioning template' do
+    template = FactoryBot.build(:provisioning_template, template: "<%= I WILL RAISE AN ERROR %>")
```

Comment:
I love this template text :)

It reminded me of https://github.com/voxpupuli/puppet-syntax/commit/ceb817621b10c621df6d3c665060cfe4ed44fc0f#diff-912e42831b0b6555ff7a04b9089b9451a35425673939995ea6282ecd780c20fb because in Ruby 3.4 `This is not valid Ruby` suddenly became syntactically valid.

I just verified it and on a plain Ruby 3.4 I see
```console
$ ruby
I WILL RAISE AN ERROR
-:1:in '<main>': uninitialized constant ERROR (NameError)
Did you mean?  IOError
               Errno
```

So as long as those constants aren't defined, this will work. However, you can make it a bit more invalid:
```suggestion
    template = FactoryBot.build(:provisioning_template, template: "<%= I WILL RAISE AN ERROR( %>")
```